src/components/global/requestsTable.vue

```diff
@@ -1,48 +1,142 @@
-                v-switch(v-model="singleSelect1" label="Single select" class="pa-3")
-        v-data-table(:headers="admin" :items-per-page="5" show-select class="elevation-1" :single-select="singleSelect2")
-            template(v-slot:top)
-                v-switch(v-model="singleSelect2" label="Single select" class="pa-3")
+        v-data-table.mt-10(:headers="headers" :items="requests" :items-per-page="5" show-select class="elevation-1" :single-select="singleSelect1")
+            template(v-slot:item.submission_date="{ item }")
+                td {{ formattedDate(item) }}
+            template(v-slot:item.status="{ item }")
+                v-chip(dark :color="getColor(item.status)") {{ item.status }}
+            template(v-slot:item.action="{ item }")
+                v-btn.mr-2(fab small dark color="green darken-2" @click="submit('Approved', item)")
+                    v-icon mdi-check
+                v-dialog(v-model="dialog" width="400")
+                    template(v-slot:activator="{ on, attrs }") 
+                        v-btn(fab small dark color="red lighten-1" v-bind="attrs" v-on="on")
+                            v-icon mdi-close
+                    v-card.pa-5
+                        h3.mb-5 Reason for declining?
+                        v-textarea(v-model="declineReason" outlined)
+                        v-divider
+                        v-card-actions
+                            v-spacer
+                            v-btn(text @click="dialog=false") Cancel
+                            v-btn(color="#F2594B" dark @click="submit('Declined', item)") Decline 
+                        
 </template>
 
 <script lang="ts">
 import Vue from 'vue'
 import Component from 'vue-class-component';
+import '@mdi/font/css/materialdesignicons.css';
+import { timeoffRequest } from '@/interfaces/timePickerTypes';
+import moment from 'moment';
 
 @Component({
     name: 'requestsTable',
 })
 
 export default class requestsTable extends Vue{
-    private user: string='';
-    private admin: string='';    
-
-    data() {
-        return {
-            singleSelect1: false,
-            singleSelect2: false,
-            selected: [],
-            user: [
-                { text: 'Request type' },
-                { text: 'From', value: 'From' },
-                { text: 'To', value: 'to' },
-                { text: 'Reason', value: 'reason' },
+    private user: string = '';
+    private admin: string = '';    
+    private requests: any = [];
+    private singleSelect1: boolean = false;
+    private selected: any = [];
+    private headers: any = [];
+    private dialog: boolean = false;
+    private declineReason: string= '';
+    private modifiedRequests: any = [];
+
+    submit(status: string, item: any) {
+        this.$axios.patch('/timeoff/'+item.id+'/review_user_timeoff_request/', {
+            status: status,
+            reason: this.declineReason,
+        })
+        .then(function (response: any) {
+			console.log(response);
```

Comment:
Remove the log statement. Also something should happen in the response. The request should be removed from the table or the status updated.